odoo/addons/base/models/qweb.py

```diff
@@ -226,166 +85,156 @@ def update(self, *args, **kwargs):
     def __hash__(self):
         return hash(frozenset((key, freehash(val)) for key, val in self.items()))
 
+unsafe_eval = eval
 
-class MarkupSafeBytes(bytes):
-    __slots__ = ()
-
-    def __new__(cls, source):
-        assert isinstance(source, bytes)
-        return super().__new__(cls, source)
+_FORMAT_REGEX = re.compile(r'(?:#\{(.+?)\})|(?:\{\{(.+?)\}\})') # ( ruby-style )|(  jinja-style  )
+_VARNAME_REGEX = re.compile(r'\W')
 
-    def __html__(self):
-        return self.decode()
-
-    def __str__(self):
-        raise NotImplementedError(
-            "Bytes should not be stringified, only repr'd (tried to stringify %s)" % reprlib.repr(self)
-        )
-
-    def decode(self, encoding='utf-8', errors='strict'):
-        return Markup(super().decode(encoding, errors))
 
 ####################################
 ###             QWeb             ###
 ####################################
 
 
 class QWeb(object):
-    _empty_line = re.compile(r'\n\s*\n')
     __slots__ = ()
 
     _void_elements = frozenset([
         'area', 'base', 'br', 'col', 'embed', 'hr', 'img', 'input', 'keygen',
         'link', 'menuitem', 'meta', 'param', 'source', 'track', 'wbr'])
     _name_gen = count()
+    _available_objects = dict({k:v for k, v in vars(builtins).items() if k[0] != '_'})
```

Comment:
What is the point of calling `dict` on a `dict`?  And `startswith` is more readable.
```suggestion
    _available_objects = {k: v for k, v in vars(builtins).items() if not k.startswith('_')}
```